app.rb

```diff
@@ -1,8 +1,49 @@
 require 'sinatra/base'
+require 'sinatra/reloader'
+
 class RockPaperScissors < Sinatra::Base
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  enable :sessions
+
+  get '/' do
+    erb :index
+  end
+
+  post '/name' do
+    p params
+    session[:player_name] = params[:player_name]
```

Comment:
Rather than using session storage, you could create a `Player` class. When the user enters their name, you could create an instance of the Player class and pass it the users name. Creating a Player class would also help when dealing with the player move - `move` could be an attribute on the `Player` class, which is set in the `/play` route. This would mean you could use a single view to set the players move.

You could also have a second class for the Game, or for the Computer move, which you could use to randomly decide the computers move, and also for the logic of deciding who the winner is. Your routing and logic is good, but by placing them in classes, that would bring the code into using OOD!

Reply:
That's a helpful suggestion, thanks. I think I hadn't understood where a separate Class, with attributes, could help here, but reading your code made this clearer.